### PR TITLE
Fix `.locale.Tr` function not found in delete modal

### PR DESCRIPTION
**Pull request by delvh**
Caught by @wxiaoguang in https://github.com/go-gitea/gitea/pull/23337#issuecomment-1467317742.

Additionally, there were three instances that have the same content as `templates/base/deletion_modal_actions.tmpl`, but that are not intended to delete something.
Instead of renaming the template above, these instances were simply re-hard-coded again.
Renaming/improving the template above is left for future PRs.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
I guess this change equals before. Because in this template, `.xx` is `$.xx`

**Comment by wxiaoguang** [comment]
Just have a try, I see the error message: 

`template: base/delete_modal_actions:4:5: executing "base/delete_modal_actions" at <$.locale.Tr>: can't evaluate field locale in type string`,

 it's `$.locale.Tr` now 😂

**Comment by delvh** [comment]
As far as I've understood Go templates (please correct me if I misunderstood something):
`.something` is the attribute `x.something` when you use `{{template "<another>" x}}`.
`$.something` is the global value set with `ctx.Data["something"]`.
Is that correct, or is there yet another thing I have to learn about the templates?

**Comment by wxiaoguang** [comment]
In a template, `$.xx` is accessing the root scope. `.xx` is accessing current `range/with` scope.

So, if we write `$.xx` or `.xx` in top scope , they are the same.

I guess the problem is caused by that the sub-template is called inside a `range` scope by `{{template "actions" .}}`. Actually, it needs to access top scope to get the `locale`: try to use `{{template "actions" $}}`?

**Comment by delvh** [comment]
@wxiaoguang unfortunately, I haven't ever been able to see the dialogue.
So, could you please tell me if it now works?

**Comment by wxiaoguang** [comment]
I just navigated to the page http://localhost:3000/user/settings/repos  , let me see ....

**Comment by wxiaoguang** [comment]
I guess the fix could be like this:


```diff
diff --git a/templates/user/settings/repos.tmpl b/templates/user/settings/repos.tmpl
index 2e107ca7f..d1cb60225 100644
--- a/templates/user/settings/repos.tmpl
+++ b/templates/user/settings/repos.tmpl
@@ -50,7 +50,7 @@
                                                                                                                {{$.CsrfTokenHtml}}
                                                                                                                <input type="hidden" name="id" value="{{$dir}}">
                                                                                                                <input type="hidden" name="action" value="adopt">
-                                                                                                               {{template "base/delete_modal_actions" .}}
+                                                                                                               {{template "base/delete_modal_actions" $}}
                                                                                                        </form>
                                                                                                </div>
                                                                                        {{end}}
@@ -68,7 +68,7 @@
                                                                                                                {{$.CsrfTokenHtml}}
                                                                                                                <input type="hidden" name="id" value="{{$dir}}">
                                                                                                                <input type="hidden" name="action" value="delete">
-                                                                                                               {{template "base/delete_modal_actions" .}}
+                                                                                                               {{template "base/delete_modal_actions" $}}
                                                                                                        </form>
                                                                                                </div>
                                                                                        {{end}}
```

**Comment by wxiaoguang** [comment]
Hmm , I think I know your question know .... some pages are for special usage, for example, adopt repositories, adopt lfs files, etc. You need to have some unadopted repositories to make Gitea render that UI.

I did a global search, I can find there are also similar problems in `templates/admin/repo/unadopted.tmpl` : `{{range}}` => `{{template "actions" .}}`.

So I guess there are 4 lines need to be changed: 2 lines in `unadopted.tmpl`, 2 lines in `templates/user/settings/repos.tmpl`


----

And I find one more possible problem, the template name `delete_modal_actions` is not accurate, for example, in the `unadopted.tmpl`, these yes/no buttons are also used to confirm to adopt, but not to delete.

**Comment by delvh** [comment]
Yep, and I agree that all three other times, `delete` is not the correct method, so I de-converted these instances now.
(Another approach would be to rename the dialogue, but I would leave that for another PR of its own)

**Review by techknowlogick — APPROVED** [review]

File: templates/admin/repo/unadopted.tmpl [hunk 1]
```diff
@@ -61,7 +70,16 @@
...
+												<div class="actions">
+													<button class="ui red basic inverted cancel button">
+														{{svg "octicon-trash" 16 "gt-mr-2"}}
+														{{$.locale.Tr "modal.no"}}
+													</button>
+													<button class="ui green basic inverted ok button">
+														{{svg "octicon-check" 16 "gt-mr-2"}}
+														{{$.locale.Tr "modal.yes"}}
+													</button>
+												</div>
```
Comment (silverwind) [inline templates/admin/repo/unadopted.tmpl:82]: Why not make another subtemplate with the trash icon? 3 repetitions is already worthy of a subtemplate.

Comment (wxiaoguang) [reply templates/admin/repo/unadopted.tmpl:82]: It just need a rename.

Rename the `delete_modal_actions` to `modal_actions_yes_no`

Comment (delvh) [reply templates/admin/repo/unadopted.tmpl:82]: I wanted to postpone this for a future PR, but yeah, we can of course do it in this one too…

Comment (wxiaoguang) [reply templates/admin/repo/unadopted.tmpl:82]: Either is  fine, fix bugs first 😁

**Review by lafriks — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 (for merge queue)

**Pull request merged**